src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -2731,6 +3571,78 @@ public S(int arg)
             Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
         }
 
+        [Theory, CombinatorialData]
+        public void InStruct_Cycle(bool firstIsSemi, bool secondIsSemi, bool bindS1First)
+        {
+            var firstAccessor = firstIsSemi ? "get => field;" : "get;";
+            var secondAccessor = secondIsSemi ? "get => field;" : "get;";
+            var source1 = $@"
+struct S1
+{{
+    public S1() {{ }}
+    public S2 P {{ {firstAccessor} }}
```

Comment:
Consider testing a version where these are static properties, to ensure that everything is fine there.

Reply:
Test is added